contracts/axelar-gateway/src/event.rs

```diff
@@ -25,9 +25,15 @@ pub(crate) fn approve_message(env: &Env, message: Message) {
         .publish(topics, (message.source_chain, message.source_address));
 }
 
-pub(crate) fn execute_contract_call(env: &Env, message_id: String) {
-    let topics = (symbol_short!("executed"), message_id);
-    env.events().publish(topics, ());
+pub(crate) fn execute_contract_call(env: &Env, message: Message) {
+    let topics = (
+        symbol_short!("executed"),
+        message.message_id,
```

Comment:
@ahramy As per Stellar [docs](https://developers.stellar.org/docs/build/guides/events/publish), event topics can't contain arbitrarily length data such as a Vec, Bytes etc. (similar to solidity). They don't mention `String` there but I'm wondering if that could potentially cause an issue (in the future, since it seems to work atm).

Reply:
@milapsheth I asked a question and got an answer, confirmed that there's no limits for topic types.

> dmytro_stellar: sorry, the doc seems incorrect/outdated. there are no limits for topic types or even count. the only limitation is that a transaction can't emit more than 8 KB of events combined. that said if you have too many/too complex topics, in theory you might have issues with indexers. that said, I don't know if there are any such restrictions in reality

Reply:
Since they said
> there are no limits for topic types or even count.
let's use the whole message for the topic?

@sdaveas Will you need the message fields to separated in the topics tuple, or a single element tuple with the full message is fine?

Reply:
it doesn't make a big difference for us because we track every event emitted by the contract, and we don't use topic attributes in the event filter. We want to consume the whole event to construct the whole message, so if the message is partially in topic and values of the event, that's fine. Ideally it'd be the same for approve and execute so that we can reuse our parsing code, but if there's a good reason to diverse, that's fine also.